### PR TITLE
Support Infrastructure Encryption - secondary layer of encryption at …

### PR DESCRIPTION
**Pull request by tonykjose**
…REST for accounts.

Add a boolean property to Microsoft.Storage/storageAccounts resource under encryption attributes to opt-in for a secondary layer of encryption for data at rest.

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/storage.json [hunk 1]
```diff
@@ -1626,6 +1626,11 @@
           },
           "default": "Microsoft.Storage"
         },
+        "requireInfrastructureEncryption": {
+          "type": "boolean",
```
Comment (Juliehzl) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/storage.json:1630]: As you shown in examples, it looks there is a default value `false` in server side for the property, right?

**Review by KrisBash — APPROVED** [review]

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/storage.json [hunk 2]
```diff
@@ -1626,6 +1626,11 @@
...
+          "x-ms-client-name": "RequireInfrastructureEncryption",
+          "description": "A boolean indicating whether or not the service applies a secondary layer of encryption with platform managed keys for data at rest."
```
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/storage.json:1632]: Is there a feature spec this it? 
What's the difference of the storage account when set the value to true or false? And if user don't set it, what's the default value?
Does the property only support on create, but not support on update account?

File: specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/storage.json (hunk 1 above)
Comment (tonykjose) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2019-06-01/storage.json:1630]: This defaults to null. Corrected the example.

**Review by Juliehzl — APPROVED** [review]

**Comment by Juliehzl** [comment]
@tonykjose Can you help confirm the breaking changes detected in CI system? If you are fine with such changes, I will help you merge it.

**Pull request merged**